core/domain/exp_jobs_one_off_test.py

```diff
@@ -264,6 +263,7 @@ def test_first_published_time_of_exploration_that_is_unpublished(self):
         exploration = self.save_new_valid_exploration(
             self.EXP_ID, self.owner_id, end_state_name='End')
         rights_manager.publish_exploration(self.owner_id, self.EXP_ID)
+        self.logout()
```

Comment:
Actually you don't need the login and logout here. You only need it when you're fetching a web page using self.testapp.get or get_json or post_json etc.